DOL.WHD.Section14c.Business/Helper/ApplicationDocumentHelper.cs

```diff
@@ -1,47 +1,151 @@
+                //exclude \r\n from html string
+                string validation = "^(?:[\r\n])*$";
+                if (!Regex.IsMatch(htmlString, validation))
+                {
+                    htmlTemplates.Add(htmlString);
+                }                
+            }
 
             // Prepare attachemnt for PDF generation
             var applicationAttachmentsData = _attachmentService.PrepareApplicationContentsForPdfConcatenation(
                                             getApplicationAttachments,
-                                            applicationHtml);
+                                            htmlTemplates);
 
             return applicationAttachmentsData;
         }
+
+        /// <summary>
+        /// Set application submission related object for pdf generation
+        /// </summary>
+        /// <param name="application">
+        /// Application Submission
+        /// </param>
+        private void LoadApplicationData(ref ApplicationSubmission application)
+        {
+            // Load WIOAWorker related data
+            if (application.WIOA != null && application.WIOA.WIOAWorkers != null)
+            {                
+                var workers = new List<WIOAWorker>();
+                foreach (var worker in application.WIOA.WIOAWorkers)
+                {
+                    worker.WIOAWorkerVerified = _responseService.GetResponseById(worker.WIOAWorkerVerifiedId.ToString());
+                    workers.Add(worker);
+                };
+                application.WIOA.WIOAWorkers = workers;
+            }
+            // Load work site related data
+            if (application.WorkSites != null)
+            {                
+                var workSites = new List<WorkSite>();
+                foreach (var wSite in application.WorkSites)
+                {
+                    wSite.WorkSiteType = _responseService.GetResponseById(wSite.WorkSiteTypeId.ToString());                    
+                    wSite.Employees = LoadEmployeeData(wSite.Employees);
+                    workSites.Add(wSite);
+                };
+                application.WorkSites = workSites;
+            }
+            // Load application establishment types related data
+            if (application.EstablishmentType != null)
+            {
+                var applicationSubmissionEstablishmentTypes = new List<ApplicationSubmissionEstablishmentType>();
+                foreach(var type in application.EstablishmentType)
+                {
+                    type.EstablishmentType = _responseService.GetResponseById(type.EstablishmentTypeId.ToString());
+                    applicationSubmissionEstablishmentTypes.Add(type);
+                }
+                application.EstablishmentType = applicationSubmissionEstablishmentTypes;
+            }
+            // Load Employer Providing Facilities Deduction types related data
+            if (application.Employer?.ProvidingFacilitiesDeductionType != null)
```

Comment:
The way you're checking nested properties is inconsistent between here and line 83.  I'm okay with either way, but it should be the same.  E.g., either line 83 should change to:
```C#
if (application.WIOA?.WIOAWorkers != null)
```
(to be honest, I think this one looks nicer)

or line 117 should change to:
```C#
if (application.Employer != null && application.Employer.ProvidingFacilitiesDeductionType != null)
```